src/main/java/com/mathworks/ci/MatlabInstallation.java

```diff
@@ -0,0 +1,77 @@
+package com.mathworks.ci;
+
+import hudson.CopyOnWrite;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.EnvironmentSpecific;
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import hudson.slaves.NodeSpecific;
+import hudson.tools.ToolDescriptor;
+import hudson.tools.ToolInstallation;
+import hudson.tools.ToolProperty;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import javax.annotation.CheckForNull;
+import net.sf.json.JSONObject;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+public class MatlabInstallation extends ToolInstallation implements EnvironmentSpecific<MatlabInstallation>, NodeSpecific<MatlabInstallation> {
+    private static final long serialVersionUID = 1L;
+
+    @DataBoundConstructor
+    public MatlabInstallation(String name, @CheckForNull String home, List<? extends ToolProperty<?>> properties) {
+        super(Util.fixEmptyAndTrim(name), Util.fixEmptyAndTrim(home), properties);
+    }
+
+    @Override public MatlabInstallation forEnvironment(EnvVars envVars) {
+        return new MatlabInstallation(getName(), envVars.expand(getHome()), getProperties().toList());
+    }
+
+    @Override
+    public MatlabInstallation forNode(Node node, TaskListener log) throws IOException, InterruptedException {
+        return new MatlabInstallation(getName(), translateFor(node, log), getProperties().toList());
+    }
+
+    @Override
+    public void buildEnvVars(EnvVars env) {
+        String pathToExecutable = getHome() + "/bin";
```

Comment:
Should this always be "/bin"? Wouldn't it be "\bin" on Windows? Or does Jenkins take care of that automatically?

Reply:
Yes, Jenkins (or windows) takes care of it. Currently, this class does not have access to build context hence extracting OS platform information is tricky.